.github/workflows/nightly.yaml

```diff
@@ -24,7 +24,75 @@ jobs:
     steps:
       - name: Checkout
         uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3.3.0
-        # TODO(Tyler): Iron out ref for tests
+
+      - name: Cache tool downloads
+        uses: actions/cache@627f0f41f6904a5b1efbaed9f96d9eb58e92e920 # v3.2.4
+        with:
+          path: /tmp/plugins_testing_download_cache
+          # No need to key on trunk version unless we change how we store downloads.
+          key: trunk-${{ runner.os }}
+
+      - name: Delete cache
+        # For now, avoid deleting cache on pull request changes to nightly. This improves PR experience.
+        if: env.TRIGGER != 'pull_request'
+        run: |
+          if [ -d "/tmp/plugins_testing_download_cache" ]
+          then
+            tmp_dir=/tmp/${GITHUB_RUN_ID}-${GITHUB_RUN_NUMBER}-${GITHUB_RUN_ATTEMPT}
+            mv "/tmp/plugins_testing_download_cache" ${tmp_dir}
+            chmod -R u+w ${tmp_dir}
+            rm -rf ${tmp_dir}
+          fi
+        shell: bash
+
+      - name: Linter Tests
+        uses: ./.github/actions/linter_tests
+        with:
+          linter-version: ${{ matrix.linter-version }}
+
+  # Run tests against all linters for snapshots and latest version as they exist in latest release
+  # This job is used to update the list of validated versions
+  linter_tests_release:
+    name: Plugin Tests
+    runs-on: ${{ matrix.os }}
+    timeout-minutes: 60
+    strategy:
+      fail-fast: false
+      matrix:
+        linter-version: [Snapshots, Latest]
+        os: [ubuntu-latest, macos-latest]
+    steps:
+      - name: Retrieve git history
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      # This assumes that any changes on main since the last release are backwards compatible
+      # in terms of allowing existing linter tests to run.
+      - name: Preserve test runner latest behavior
```

Comment:
This is required at least for now, since the latest release doesn't have any of the upload scripts